circuits/cpp/src/aztec3/circuits/kernel/private/common.cpp

```diff
@@ -0,0 +1,205 @@
+#include "init.hpp"
+
+#include "aztec3/circuits/abis/contract_deployment_data.hpp"
+#include "aztec3/circuits/abis/function_data.hpp"
+#include "aztec3/circuits/abis/kernel_circuit_public_inputs.hpp"
+#include "aztec3/circuits/abis/new_contract_data.hpp"
+#include "aztec3/circuits/abis/private_kernel/private_call_data.hpp"
+#include "aztec3/circuits/hash.hpp"
+#include "aztec3/constants.hpp"
+#include "aztec3/utils/array.hpp"
+#include "aztec3/utils/dummy_composer.hpp"
+
+using DummyComposer = aztec3::utils::DummyComposer;
+
+using aztec3::circuits::abis::ContractDeploymentData;
+using aztec3::circuits::abis::ContractLeafPreimage;
+using aztec3::circuits::abis::FunctionData;
+using aztec3::circuits::abis::KernelCircuitPublicInputs;
+using aztec3::circuits::abis::NewContractData;
+
+using aztec3::utils::array_push;
+using aztec3::utils::is_array_empty;
+using aztec3::utils::push_array_to_array;
+using DummyComposer = aztec3::utils::DummyComposer;
+using CircuitErrorCode = aztec3::utils::CircuitErrorCode;
+using aztec3::circuits::abis::private_kernel::PrivateCallData;
+
+namespace aztec3::circuits::kernel::private_kernel {
+
+void common_validate_call_stack(DummyComposer& composer, PrivateCallData<NT> const& private_call)
```

Comment:
The functions in this file and in the kernels should really have top-level comments. That can be done in a separate PR IMO since they were uncommented before.

Reply:
Added a TODO for this.